doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst

```diff
@@ -98,6 +100,13 @@ To enable the FFS support, set the following configuration options to meet the A
 * :kconfig:option:`CONFIG_CHIP_DEVICE_TYPE` to the appropriate value, depending on the device used.
   The value must be compliant with the Matter Device Type Identifier.
 
+Every Matter device must use an unique device identifier for rotating device identifier calculation purpose.
+In case of using factory data, the unique identifier is read from the factory data partition.
+In other cases, the value set through the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option is used.
+You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option to ``y``.
+The auto-generation mode will set the identifier to a random value.
```

Comment:
```suggestion
You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the :kconfig:option:`CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig option to ``y``.
In case of using factory data, the unique identifier is read from the factory data partition.
Otherwise, the auto-generation mode will set the identifier to a random value.
```
Mulled that around in my head for a bit more. Check that no relevant information was lost in the reordering.

Reply:
To be honest I think the previous version was better. The information we want to include is that you can store the unique id in the factory data or use hardcoded version from the firmware. Next, if you selected factory data, there are again two ways to go. You can define your own value and store it in the factory data, or you don't care about it and choose the auto-generation that will use some random value and save it in the factory data.

The suggested version starts from: 

> You can choose your own unique identifier value to be stored in the factory data or select the auto-generation mode by setting the ..

and in my opinion it informs that I can store id in the factory data or select auto-generation, so I would assume the latter will not use factory data, which is not true. 

What do you think?

Reply:
Ah, so the auto-generation also stores the generated identifier in the factory data, and that is where it is read from regardless?

Reply:
Yes, btw. I think I made some mistake and used the same kconfig name two times. Let me explain it one more time. the user can choose between:
a. Keeping unique id, as a constant value in the firmware (passed through `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` Kconfig)
b. Keeping unique id in factory data

In case of selecting b, we have another two options:
a. Select auto-generation that will use random unique id value and put it into the storage (`CONFIG_CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID` equal to `y`, then `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` is ignored)
b. Force using own unique id value that will be put into the storage (`CONFIG_CHIP_DEVICE_GENERATE_ROTATING_DEVICE_UID` equal to `n`, then `CONFIG_CHIP_DEVICE_ROTATING_DEVICE_UID` is used)